src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/tests/test_acr_commands_mock.py

```diff
@@ -0,0 +1,274 @@
+            headers=_get_authorization_header('username', 'password'))
+
+        # Delete manifest with digest
+        acr_repository_delete(cmd, 'testregistry', 'testrepository', manifest='sha256:c5515758d4c5e1e838e9cd307f6c6a0d620b5e07e6f927b07d05f6d12a1ac8d7', yes=True)
+        mock_requests_delete.assert_called_with(
+            'https://testregistry.azurecr.io/v2/testrepository/manifests/sha256:c5515758d4c5e1e838e9cd307f6c6a0d620b5e07e6f927b07d05f6d12a1ac8d7',
+            headers=_get_authorization_header('username', 'password'))
+
+    @mock.patch('azure.cli.core._profile.Profile.get_refresh_token', autospec=True)
+    @mock.patch('azure.cli.command_modules.acr._docker_utils.get_registry_by_name', autospec=True)
+    @mock.patch('requests.post', autospec=True)
+    @mock.patch('requests.get', autospec=True)
+    def test_get_docker_credentials(self, mock_requests_get, mock_requests_post, mock_get_registry_by_name, mock_get_refresh_token):
+        cmd = mock.MagicMock()
+        cmd.cli_ctx = TestCli()
+
+        registry = Registry(location='westus', sku=Sku('Standard'))
+        registry.login_server = 'testregistry.azurecr.io'
+        mock_get_registry_by_name.return_value = registry, None
+
+        # Set up challenge response
+        challenge_response = mock.MagicMock()
+        challenge_response.headers = {
+            'WWW-Authenticate': 'Bearer realm="https://testregistry.azurecr.io/oauth2/token",service="testregistry.azurecr.io"'
+        }
+        challenge_response.status_code = 401
+        mock_requests_get.return_value = challenge_response
+
+        # Set up refresh/access token response
+        refresh_token_response = mock.MagicMock()
+        refresh_token_response.headers = {}
+        refresh_token_response.status_code = 200
+        refresh_token_response.content = json.dumps({
+            'refresh_token': 'testrefreshtoken',
+            'access_token': 'testaccesstoken'}).encode()
+        mock_requests_post.return_value = refresh_token_response
+
+        # Set up AAD token with both refresh and access tokens
+        mock_get_refresh_token.return_value = None, 'aadrefreshtoken', 'aadaccesstoken', 'testtenant'
+        get_login_credentials(cmd.cli_ctx, 'testregistry', None, None, None)
+        mock_requests_get.assert_called_with('https://testregistry.azurecr.io/v2/')
+        mock_requests_post.assert_called_with(
+            'https://testregistry.azurecr.io/oauth2/exchange',
+            urlencode({
+                'grant_type': 'access_token_refresh_token',
+                'service': 'testregistry.azurecr.io',
+                'tenant': 'testtenant',
+                'access_token': 'aadaccesstoken',
+                'refresh_token': 'aadrefreshtoken'
+            }),
+            headers={'Content-Type': 'application/x-www-form-urlencoded'})
+
+        get_access_credentials(cmd.cli_ctx, 'testregistry', None, None, None, 'testrepository')
```

Comment:
Just to verify, does this eventually call the _get_credentials method where the login regression occurred?

Reply:
Yes. Both `get_access_credentials` and `get_login_credentials` call `_get_credentials`.